bee-api/bee-rest-api/src/endpoints/routes/api/v2/submit_message.rs

```diff
@@ -78,40 +81,39 @@ pub(crate) async fn submit_message<B: StorageBackend>(
     rest_api_config: RestApiConfig,
     protocol_config: ProtocolConfig,
 ) -> Result<impl Reply, Rejection> {
-    let network_id_v = &value["networkId"];
-    let parents_v = &value["parentMessageIds"];
-    let payload_v = &value["payload"];
-    let nonce_v = &value["nonce"];
+    let protocol_version_json = &value["protocolVersion"];
+    let parents_json = &value["parentMessageIds"];
+    let payload_json = &value["payload"];
+    let nonce_json = &value["nonce"];
 
-    // parse the values, take care of missing fields, build the message and submit it to the node for further
-    // processing
+    // Tries to build a `Message` from the given JSON.
+    // If some fields are missing, it tries to auto-complete them.
 
-    let network_id = if network_id_v.is_null() {
-        network_id.1
-    } else {
-        network_id_v
-            .as_str()
-            .ok_or_else(|| {
-                reject::custom(CustomRejection::BadRequest(
-                    "invalid network id: expected an u64-string".to_string(),
-                ))
-            })?
-            .parse::<u64>()
-            .map_err(|_| {
-                reject::custom(CustomRejection::BadRequest(
-                    "invalid network id: expected an u64-string".to_string(),
-                ))
-            })?
-    };
+    if !protocol_version_json.is_null() {
+        let parsed = protocol_version_json.as_u64().ok_or_else(|| {
```

Comment:
I suggest renaming to `parsed_version` or even `parsed_protocol_version` to increase readability of the below code without having to come back to this line 93 to understand what `parsed` actually contains. Ignore that advise if you think this is too obvious, because admittedly I am not too familiar with this crate :slightly_smiling_face:

Reply:
Done 👍